### PR TITLE
Add full coverage for smart variables in UI

### PR DESCRIPTION
**Pull request by oshtaier**
Behold the might of <del>Stormwind</del> <del>Orgrimmar</del>ughm... of smart variables for UI :)

Latest Downstream(6.2.x):
```
nosetests tests/foreman/ui/test_variables.py
............................................
----------------------------------------------------------------------
Ran 44 tests in 2576.700s

OK
```
Latest Upstream:
65% of pass rate. It is really hard to determine whether style used in the application is the one that will be used for 'black' edition or it is specific of 'blue' edition. + It is really hard to work further with such huge bunch of code. Let's merge it and I will work on fixing right away. Functionality will be the same, some changes to elements will be required only.

### CODE REVIEW

File: robottelo/ui/locators/menu.py [hunk 1]
```diff
@@ -5,10 +5,8 @@
 from .model import LocatorDict
 
 NAVBAR_PATH = (
-    '//div[contains(@class,"navbar") '
-    'and contains(@class,"persist-header") and '
-    'not(contains(@style, "display: none")) and '
-    'not(contains(@style, "display:none"))]'
+    '//div[contains(@class,"navbar-inner") and '
```
Comment (oshtaier) [inline robottelo/ui/locators/menu.py]: That locator doesn't work for all situations and sometimes after some actions it starts to point to two elements instead one and as a consequence I got a fail for bunch of my test cases. That new one works fine as you can see in my test results for this PR.
Verified both for downstream and upstream builds

File: tests/foreman/api/test_classparameters.py [hunk 2]
```diff
@@ -138,6 +138,12 @@ def setUpClass(cls):
             raise RuntimeError('There are not enough smart class parameters to'
                                ' work with in provided puppet class')
 
+    @classmethod
+    def tearDownClass(cls):
```
Comment (oshtaier) [inline tests/foreman/api/test_classparameters.py]: Moving `tearDownClass` block into the right place.

File: tests/foreman/ui/test_variables.py [hunk 3]
```diff
@@ -180,336 +451,398 @@ def test_negative_update_type(self):
...
+                    )
+                    self.smart_variable.click(self.smart_variable.search(name))
+                    value = self.smart_variable.wait_until_element(
+                        locators['smart_variable.default_value']).text
+                    self.assertEqual(value, initial_value)
 
     @run_only_on('sat')
-    @stubbed()
-    @tier3
-    def test_negative_validate_default_value_required_checkbox(self):
```
Comment (oshtaier) [inline tests/foreman/ui/test_variables.py]: `Required` checkbox functionality is not supported by Smart Variables and actually it is not even present on UI

File: tests/foreman/ui/test_variables.py [hunk 4]
```diff
@@ -180,336 +451,398 @@ def test_negative_update_type(self):
...
-
-        @caseautomation: notautomated
-
-        @CaseLevel: System
-        """
-
-    @run_only_on('sat')
-    @stubbed()
-    @tier3
-    def test_positive_validate_default_value_required_checkbox(self):
```
Comment (oshtaier) [inline tests/foreman/ui/test_variables.py]: `Required` checkbox functionality is not supported by Smart Variables and actually it is not even present on UI

File: tests/foreman/ui/test_variables.py [hunk 5]
```diff
@@ -180,336 +451,398 @@ def test_negative_update_type(self):
...
-
-        @caseautomation: notautomated
-
-        @CaseLevel: System
-        """
-
-    @run_only_on('sat')
-    @stubbed()
-    @tier3
-    def test_negative_validate_matcher_value_required_checkbox(self):
```
Comment (oshtaier) [inline tests/foreman/ui/test_variables.py]: `Required` checkbox functionality is not supported by Smart Variables and actually it is not even present on UI

File: tests/foreman/ui/test_variables.py [hunk 6]
```diff
@@ -180,336 +451,398 @@ def test_negative_update_type(self):
...
-
-        @caseautomation: notautomated
-
-        @CaseLevel: System
-        """
-
-    @run_only_on('sat')
-    @stubbed()
-    @tier3
-    def test_positive_validate_matcher_value_required_checkbox(self):
```
Comment (oshtaier) [inline tests/foreman/ui/test_variables.py]: `Required` checkbox functionality is not supported by Smart Variables and actually it is not even present on UI

File: tests/foreman/ui/test_variables.py [hunk 7]
```diff
@@ -780,122 +1347,153 @@ def test_negative_create_matcher_avoid_duplicate(self):
...
+                ]
+            )
+            self.assertIsNotNone(self.smart_variable.search(name))
+            output = yaml.load(self.hosts.get_yaml_output(self.host_name))
+            self.assertEqual(output['parameters'][name], [70, 80, 90, 100])
 
     @run_only_on('sat')
-    @stubbed()
-    @tier3
-    def test_positive_enable_merge_overrides_default_checkboxes(self):
```
Comment (oshtaier) [inline tests/foreman/ui/test_variables.py]: Such scenario covered a lot of times in other cases

File: tests/foreman/ui/test_variables.py [hunk 8]
```diff
@@ -780,122 +1347,153 @@ def test_negative_create_matcher_avoid_duplicate(self):
...
+            )
+            self.assertFalse(
+                self.smart_variable.validate_checkbox(name, 'Merge Overrides'))
+            self.assertFalse(
+                self.smart_variable.validate_checkbox(name, 'Merge Default'))
 
     @run_only_on('sat')
-    @stubbed()
-    @tier3
-    def test_positive_enable_avoid_duplicates_checkbox(self):
```
Comment (oshtaier) [inline tests/foreman/ui/test_variables.py]: Functionality is covered in other cases

File: tests/foreman/ui/test_variables.py [hunk 9]
```diff
@@ -913,62 +1511,66 @@ def test_positive_create_override_from_attribute(self):
...
+                self.host_name, name, gen_string('alpha'))
+            self.assertIsNotNone(
+                self.hosts.wait_until_element(locators['host.override_error']))
+            self.assertTrue(self.smart_variable.validate_smart_variable(
+                name, 'overrides_number', '0'))
 
     @run_only_on('sat')
-    @stubbed()
-    @tier1
-    def test_negative_create_override_from_attribute_required_checked(self):
```
Comment (oshtaier) [inline tests/foreman/ui/test_variables.py]: `Required` checkbox functionality is not supported by Smart Variables and actually it is not even present on UI

**Comment by renzon** [comment]
ACK, What a big PR!

**Comment by oshtaier** [comment]
@abalakh please review

**Comment by oshtaier** [comment]
ok, nvm

**Pull request merged**